src/rhsmlib/services/entitlement.py

```diff
@@ -330,6 +376,21 @@ def filter_pool_by_service_level(pool_data):
         if service_level is not None:
             available_pools = list(filter(filter_pool_by_service_level, available_pools))
 
+        # When pagination result of available pools is requested, then reduce too long list
+        if items_per_page > 0:
```

Comment:
I think we should revisit paging in general across our tooling. This is a good solution for now.
In the future I think the subscription tooling would benefit from improvements to the implementation of paging in Candlepin to ensure consistency for paging results from a given query and to reduce the memory overhead.

If we could be sure that we'll get responses that don't miss items we could make use of candlepin's paging APIs. The only other thing to consider is RTT (as you have here). Making use of Candlepin's paging could reduce performance significantly if the RTT is high. Could we implement a hybrid approach? If the RTT is past some limit, use local paging and grab the whole set of pools. If RTT is below that number, grab directly from candlepin always using candlepin's paging?